src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -127,7 +127,7 @@
 
     private static String sBlobVersion;
```

Comment:
The same problem that you currently fixing:
One 1 set Key and MacKey and version to VERSION_ANDROID_KEY_STORE
Another app sets another Key and MacKey but for some reason sets version to VERSION_USER_DEFINED
Then if you're using static version you won't be able to decrypt anything with first Key, because it thinks it has wrong version.